server/devdocs/selectors/index.js

```diff
@@ -23,47 +29,105 @@ const SELECTORS_DIR = path.resolve( __dirname, '../../../client/state/selectors'
 const router = express.Router();
 let prepareFuse;
 
-function parseSelectorFile( file ) {
-	return new Promise( ( resolve, reject ) => {
-		fs.readFile( SELECTORS_DIR + '/' + file, 'utf8', ( error, contents ) => {
-			if ( error ) {
-				return reject( error );
+const jsParse = contents => {
+	try {
+		return espree.parse( contents, {
+			attachComment: true,
+			ecmaVersion: 6,
```

Comment:
>I chose `espree` because I thought that's what `eslint` was using. I don't think it matters that much and I can change it out.

There's not a whole lot of difference between the options out there, but I expect it'd be quite painful to debug if those small differences were ever to surface. You're right that `espree` is the default parser for ESLint, but we actually substitute this with Babel's parser [in our configuration](https://github.com/Automattic/wp-calypso/blob/d0969375cca0c842676172f76f105f1fb9afa669/.eslintrc.js#L4).